### ISSUE TITLE
Can't downgrade prerelease version using find | install

### ISSUE THREAD

**Issue by rkeithhill** [PowerShell/PSResourceGet#348]
I updated to PSReadlLine 2.2.0-beta2 but on my system, it has some serious perf issues and when I start an admin Windows Terminal, none of my history is available.  So I want to downgrade to beta1.  I can remove beta2 but my first attempt to install beta1 failed:
```
> Find-PSResource psreadline -Prerelease -Version 2.2.0-beta1 | Install-PSResource
Install-PSResource: Unable to cast object of type 'System.Version' to type 'NuGet.Versioning.NuGetVersion'.
```
Seems like that should work.  I did get the following to work:
```
Install-PSResource PSReadLine -Prerelease -Version 2.2.0-beta1
```
Which makes the `find | install` failure that much more puzzling.

**Comment by SydneyhSmith** [comment]
Thanks for reporting this @rkeithhill we are able to reproduce it

**Comment by anamnavi** [comment]
Just a note for when I investigate this: if I have a non-prerelease version also installed, running his command removes that other version while also not installing the downgraded version.

**Comment by anamnavi** [comment]
This works now and will be resolved by PR #523 to be released in 3.0.12 preview release soon. You still can't install two versions side by side which have the same powershell version part (i.e 2.2.0-beta1 and 2.2.0-beta2 and 2.2.0 all have the same powershell version part. But downgrading should work:
![image](https://user-images.githubusercontent.com/30238344/140186953-c43cde2e-8c03-4697-8923-73e8b304a38e.png)
